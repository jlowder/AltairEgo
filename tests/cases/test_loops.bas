10 REM LOOP AND CONTROL FLOW TESTS
20 PRINT "=== LOOP TESTS ==="
30 REM Test basic FOR loop
40 FOR I=1 TO 5
50 PRINT "I="; I
60 NEXT I
70 REM Test FOR loop with STEP
80 FOR I=2 TO 10 STEP 2
90 PRINT "Even: "; I
100 NEXT I
110 REM Test negative STEP
120 FOR I=10 TO 1 STEP -1
130 PRINT "Countdown: "; I
140 NEXT I
150 REM Test nested FOR loops
160 FOR I=1 TO 3
170 FOR J=1 TO 2
180 PRINT "I="; I; " J="; J
190 NEXT J
200 NEXT I
210 REM Test WHILE-like behavior with GOTO
220 I=1
230 IF I>5 THEN GOTO 270
240 PRINT "While-like: "; I
250 I=I+1
260 GOTO 230
270 REM Test ON...GOTO
280 FOR X=1 TO 3
290 ON X GOTO 310, 320, 330
300 GOTO 340
310 PRINT "Case 1": GOTO 340
320 PRINT "Case 2": GOTO 340
330 PRINT "Case 3"
340 NEXT X
350 END
