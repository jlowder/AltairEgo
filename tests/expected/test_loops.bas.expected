Altair Ego: Emulating Altair BASIC 32K Rev. 3.2
OK
OK
=== LOOP TESTS ===
I= 1
I= 2
I= 3
I= 4
I= 5
Even:  2
Even:  4
Even:  6
Even:  8
Even:  10
Countdown:  10
Countdown:  9
Countdown:  8
Countdown:  7
Countdown:  6
Countdown:  5
Countdown:  4
Countdown:  3
Countdown:  2
Countdown:  1
I= 1  J= 1
I= 1  J= 2
I= 2  J= 1
I= 2  J= 2
I= 3  J= 1
I= 3  J= 2
While-like:  1
While-like:  2
While-like:  3
While-like:  4
While-like:  5
Case 1
Case 2
Case 3
OK
